webui/src/lib/api/index.js

```diff
@@ -630,12 +630,32 @@ class Objects {
     async list(repoId, ref, tree, after = "", presign = false, amount = DEFAULT_LISTING_AMOUNT, delimiter = "/") {
         const query = qs({prefix: tree, amount, after, delimiter, presign});
         const response = await apiRequest(`/repositories/${encodeURIComponent(repoId)}/refs/${encodeURIComponent(ref)}/objects/ls?` + query);
+
+        if (response.status === 404) {
+            throw new NotFoundError(`no objects found`);
```

Comment:
I think this only happens when the repo or the ref are not found - but not the path.  Perhaps best to quote the message on the error.

Reply:
I used the returned message with a fallback to a static string if it's null or undefined.